R/mzQCjson.R

```diff
@@ -0,0 +1,255 @@
+      cv_Term$accession[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+    }
+    if(substring(x,1,4)  == "name"){
+      cv_Term$name[term_index] = str_sub(x, str_locate(x, ":")[1]+2)
+
+    }
+    if(substring(x,1,4)  == "is_a"){
+      ## Connect multiple unit through commas
+      cv_Term$units[term_index] = paste(cv_Term$units[term_index], str_sub(x, str_locate(x, ":")[1]+2), seq=",")
+
+    }
+  }
+  ##Divide the units joined by commas into a list
+  cv_Term$units <- lapply(cv_Term$units, function(x){
+    if(x != ""){
+      
+      units <- unlist(strsplit(x , split = ","))
+      data <- (lapply(units, function(y){
+        cvRef <- str_sub(y, 1, str_locate(y, "\\:")[1]-1)
+        accession <- str_sub(y, 1, str_locate(y,"\\ !")[1]-1)
+        name <- str_sub(y, str_locate(y, "\\! ")[1]+2)
+        return(list(cvRef, accession, name))
+      }))
+      return(rjson::toJSON(data))
+      
+    }else{
+      data <- NA
+      return(data)
+    }
+  })
+  return(cv_Term)
+}
+
+
+#'
+#' Get all information needed in metric (cvRef, accession, name, value, unit, type of quality, input_file),
+#' type of quality and input_file act on the connection to the other two tables(inputfile_info, analysisSoftware)
+#' that belong to metadata
+#' 
+#' 
+#' 
+#' @param qcMetric A list containing all metric information
+#' @param data data frame of all cv Parameter
+#' @return data frame of all metric that appears in PTXQC
+#' 
+#' @export
+#' 
+#' 
+#' 
+readvalue <-  function(qcMetric, data){
+  runQuality = FALSE
+  setQuality = FALSE
+  new_metric = length(data$cvRef) + 1
+  for(klass in qcMetric){
+    index <- 1
+    for(metric_name in klass$qcCV){
+      ##If this metric exists in data, other information about this metric is extracted from data, 
+      ##or if not, a new line of data is added to data and a warning is given
+      if(metric_name %in% data$name) {
+        if(! is.na(data$value[which(metric_name == data$name)])) warning("Repeat metric! Each metric has only one. Please compare the data entered before with the existing data and Retain original data.")
+        else {
+          locate = which(data$name == metric_name)
+          data$value[locate] = rjson::toJSON(klass$mzQCdata[index])
+          data$quality_type[locate] = klass$quality_type[index]
+          data$input_file[locate] = klass$input_file[index]
+          
+        }
+      }else{
+        warning("There is no this metric in cv Term of mzQC, please write to the mailing list psidev-qc-dev@lists.sourceforge.net to update mzQC-metrics list.")
+        name = metric_name
+        value =  rjson::toJSON(klass$mzQCdata[index])
+        quality_type = klass$quality_type[index]
+        input_file = klass$input_file[index]
+        data[new_metric, ] <- c(NA, NA, metric_name, NA, value, quality_type, input_file)
+        new_metric = new_metric + 1
+      }
+      index = index + 1
+    }
+  }
+  ##Check whether "..Ecoli_01" or "..Ecoli_02" is included in metric's value
```

Comment:
do not check for names which are very specific to a single dataset. They will differ in the next one.
What are you trying to achieve here?

Reply:
I want to checking whether it contains "..Ecoli_01" and "..Ecoli_02" and then use that to determine whether "Raw file: Ecoli" should exist in the inputfiles of runQuality or setQuality.